### ISSUE TITLE
6.0.0-alpha.1 tries to process configuration files in ignored directories

### ISSUE THREAD

**Issue by Standard8** [eslint/eslint#11724]
**Tell us about your environment**

* **ESLint Version:** 6.0.0-alpha.1
* **Node Version:** 8.16.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

Top-level `.eslintrc.js`
```js
module.exports = {
    "parserOptions": {
        "ecmaVersion": 2017
    },
};
```
Top-level `.eslintignore`
```
test/**
```
For `test/.eslintrc.js`
```js
module.exports = {
  "extends": [
    "prettier",
  ]
};
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

No source code necessary (I had an index.js that was commented out).

```bash
./node_modules/.bin/eslint .
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ ./node_modules/.bin/eslint .

Oops! Something went wrong! :(

ESLint: 6.0.0-alpha.1.
ESLint couldn't find the config "prettier" to extend from. Please check that the name of the config is correct.
```

Normal files within ignored directories are correctly ignored, but it seems that the `.eslintrc.js` files are still processed.

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by mysticatea** [comment]
Thank you for your report.

Actually, this is intentional behavior because we change the order of loading; now ESLint loads config files before target files. The purpose of this change is to configure target files in config files. For example, https://github.com/eslint/rfcs/pull/20, https://github.com/eslint/rfcs/pull/22. With either proposal, we can add the `.eslintrc.js` as a lint target with the config file. ESLint needs to load the config file when ESLint entered into the directory while traversing in order to realize such a feature.

If you used `test` instead of `test/**` to ignore the directory, we lose the ability to unignore any files in the directory with negative file patterns, so ESLint doesn't enter the directory.

**Comment by Standard8** [comment]
Ok, I think that makes sense, thank you for the response.

If nothing else, I'd suggest this is flagged in the breaking changes, and maybe also adding an example to the [configuration section](https://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories) around ignoring files. There's probably not too many people with sub-directory imports in this fashion, but the solution wasn't immediately obvious to me.
